core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/puppeteer-utils.ts

```diff
@@ -410,6 +459,36 @@ export class BaseUser {
   getCurrentUrlWithoutParameters(): string {
     return this.page.url().split('?')[0];
   }
+
+  /**
+   * This function takes a screenshot of the page
+   * If there's no parameter for newPage, it check this.page instead of newPage.
+   * If there's no image named as the given string, it stores the screenshot with the given string in the file __image_snapshots__
+   * Otherwise, it compares the screenshot with the image named as the given string to check if they match.
+   * If they don't match, it generates an image in the file __diff_output__ to show the difference.
+   */
+  async screenshotMatch(imageName: string, newPage?: Page): Promise<void> {
+    if (process.env.MOBILE !== 'true') {
+      try {
+        const currentPage =
+          typeof newPage !== 'undefined' ? newPage : this.page;
+        await currentPage.mouse.move(0, 0);
+        await currentPage.waitForTimeout(5000);
+        expect(await currentPage.screenshot()).toMatchImageSnapshot({
+          failureThreshold: (await currentPage.$(backgroundBanner))
```

Comment:
Can you add a comment explaining where the 0.03 and 0.003 come from, and why we have different values for each? (I assume it's because of the background banner randomness -- and if this is making things annoying to test, then I'm totally fine with dropping the randomness part of it.)